### PR TITLE
Marco/trvae refactor

### PR DESCRIPTION
**Pull request by Cottoneyejoe95**
This should contain newest changes

### CODE REVIEW

**Comment by Koncopd** [comment]
@Cottoneyejoe95 ok, looks great and works fine. Though can we have an option to print the training loss also along with the validation loss?

**Comment by Cottoneyejoe95** [comment]
@Koncopd thank you! I added an option for that in newest push!

**Comment by Koncopd** [comment]
Thanks.
